core/capabilities/ccip/common/pluginconfig.go

```diff
@@ -21,30 +19,55 @@ type PluginConfig struct {
 	ContractTransmitterFactory cctypes.ContractTransmitterFactory
 	// PriceOnlyCommitFn optional method override for price only commit reports.
 	PriceOnlyCommitFn string
+	ChainRW           ChainRWProvider
+	AddressCodec      ChainSpecificAddressCodec
+	ExtraDataCodec    SourceChainExtraDataCodec
 }
 
-// PluginConfigFactory is a factory for creating PluginConfig instances.
-type PluginConfigFactory struct {
-	EVMPluginConfig    PluginConfig
-	SolanaPluginConfig PluginConfig
+// PluginServices aggregates services for a specific chain family.
+type PluginServices struct {
+	PluginConfig   PluginConfig
+	AddrCodec      AddressCodec
+	ExtraDataCodec *ExtraDataCodec
+	ChainRW        MultiChainRW
 }
 
-// NewPluginConfigFactory is a constructor for PluginConfigFactory.
-func NewPluginConfigFactory(evmPluginConfig, solanaPluginConfig PluginConfig) *PluginConfigFactory {
-	return &PluginConfigFactory{
-		EVMPluginConfig:    evmPluginConfig,
-		SolanaPluginConfig: solanaPluginConfig,
-	}
+// InitFunction defines a function to initialize a PluginConfig.
+type InitFunction func(logger.Logger, *ExtraDataCodec) PluginConfig
+
+var registeredFactories = make(map[string]InitFunction)
+
+// RegisterPluginConfig registers a plugin config factory for a chain family.
+func RegisterPluginConfig(chainFamily string, factory InitFunction) {
+	registeredFactories[chainFamily] = factory
 }
 
-// CreatePluginConfig creates a PluginConfig instance based on the chain family.
-func (f *PluginConfigFactory) CreatePluginConfig(chainFamily string) (PluginConfig, error) {
-	switch chainFamily {
-	case chainsel.FamilyEVM:
-		return f.EVMPluginConfig, nil
-	case chainsel.FamilySolana:
-		return f.SolanaPluginConfig, nil
-	default:
-		return PluginConfig{}, fmt.Errorf("unsupported chain family: %s", chainFamily)
+// GetPluginServices initializes and returns PluginServices for a chain family.
+func GetPluginServices(lggr logger.Logger, chainFamily string) (PluginServices, error) {
+	_, exists := registeredFactories[chainFamily]
+	if !exists {
+		return PluginServices{}, fmt.Errorf("unsupported chain family: %s", chainFamily)
```

Comment:
Not sure how often we'd hit this error in practice, but for easier debugging:

```suggestion
		return PluginServices{}, fmt.Errorf("unsupported chain family: %s (available: %v)", chainFamily, maps.Keys(ac.registeredAddressCodecMap))
```